src/features/logging/aggregate/index.js

```diff
@@ -0,0 +1,63 @@
+import { getInfo, getRuntime } from '../../../common/config/config'
+import { registerHandler } from '../../../common/event-emitter/register-handler'
+import { AggregateBase } from '../../utils/aggregate-base'
+import { FEATURE_NAME, LOGGING_EVENT_EMITTER_TYPES } from '../constants'
+import { Log } from '../shared/log'
+
+export class Aggregate extends AggregateBase {
+  static featureName = FEATURE_NAME
+  #agentRuntime
+  #agentInfo
+  constructor (agentIdentifier, aggregator) {
+    super(agentIdentifier, aggregator, FEATURE_NAME)
+
+    /** held logs before sending */
+    this.bufferedLogs = []
+    /** held logs during sending, for retries */
+    this.outgoingLogs = []
+
+    this.#agentRuntime = getRuntime(this.agentIdentifier)
+    this.#agentInfo = getInfo(this.agentIdentifier)
+
+    this.waitForFlags([]).then(() => {
+      /** emitted by instrument class (wrapped loggers) or the api methods directly */
+      registerHandler(LOGGING_EVENT_EMITTER_TYPES.LOG, this.handleLog.bind(this), this.featureName, this.ee)
+      this.drain()
+    })
+  }
+
+  handleLog (timestamp, message, attributes, level) {
+    this.bufferedLogs.push(
+      new Log(
+        this.#agentRuntime.timeKeeper.convertRelativeTimestamp(timestamp),
+        message,
+        attributes,
+        level
+      )
+    )
+  }
+
+  preparePayload () {
+    this.outgoingLogs.push(...this.bufferedLogs.splice(0))
+    return {
+      common: {
+        attributes: {
+          entityGuid: (this.#agentRuntime.appMetadata?.agents?.[0] || {}).entityGuid,
+          session: {
```

Comment:
More of a question, but are there actual solid commitments to incorporate logging with replay and/or traces by the UI?

Reply:
yes